Prog/vacant-lots-montgomery_YS.Rmd

```diff
@@ -0,0 +1,1257 @@
+  
+  y <- dataset %>% 
+    get_unit_variables_r(construction_efficiency,
+                       average_unit_size) %>%
+    group_by(parameters, housing_type) %>% 
+    summarize(lots = n(),
+              units_height = sum(units_height, na.rm = TRUE),
+              units_total = sum(units_total, na.rm = TRUE),
+              units_per_lot = floor(sum(units_total / lots))) %>% 
+    mutate(housing_type = as.character(housing_type))
+  
+  
+  if (dwelling_type_breakdown == TRUE) {
+   
+      bind_rows(y, mutate(x, housing_type = "Sum"))  
+  
+  } else if (dwelling_type_breakdown == FALSE) {
+      x
+    }
+}
+
+
+
+```
+    
+
+Get summary statistics for several different parameters
+
+```{r sum Rockville}
+
+get_unit_summary_r(dataset = zoningdata_rockvile1_final,
+                 construction_efficiency = .7,
+                 average_unit_size = 940,
+                 dwelling_type_breakdown = TRUE) %>% 
+  knitr::kable()
+
+
+testdataset <- get_unit_variables_r(dataset = zoningdata_rockvile1_final,
+                 construction_efficiency = .7,
+                 average_unit_size = 940) %>% 
+filter(housing_type=="SFA") %>% 
+ filter(category=="vacant") %>% 
+  filter(smallsize_flag==0) %>% 
+  select(zoning_code, propaddress, dwelling_type, lotsize_sf, height_ft, lot_coverage_max, units_height, buildable_area, living_area, units_total, units_per_floor, floors, one_unit_flag,  subset, lot_area_min, lotarea_per_unit, smallsize_flag, zone)
+
+
+map_df(c(.7, .85), ~ get_unit_summary_r(dataset = zoningdata_rockvile1_final,
+                                      construction_efficiency = .,
+                                      average_unit_size = 940,
+                                      dwelling_type_breakdown = FALSE)) %>% 
+  knitr::kable()
+
+```
+
+Look at the distribution of units for lots with more than 1 unit, to check for extreme outliers.
+
+```{r scatter Rockville, echo = FALSE}
+
+zoningdata_rockvile1_final %>% 
+  get_unit_variables_r(construction_efficiency = .7,
+                     average_unit_size = 940) %>% 
+  filter(units_total > 1) %>% 
+  ggplot() +
+  geom_point(mapping = aes(lotsize_sf, units_total),
+             alpha = .5) +
+  scale_x_continuous(labels = scales::comma,
+                     expand = expand_scale(mult = c(0, 0.002)), 
+                     limits = c(0, 300000),
+                     breaks = 0:6 * 50000) +
+  scale_y_continuous(expand = expand_scale(mult = c(0, 0.002)), 
+                     limits = c(0, 300),
+                     breaks = 0:7 * 50) +
+  scatter_grid() +
+  labs(x = "Lot size (square feet)", y = "Total units") +
+  theme(plot.margin = margin(t = 20, r = 20))
+
+
+```
+
+
+Look more closely at vacant lots < 50,000 SF
+
+```{r scatter-small Rockville, echo = FALSE}
+
+zoningdata_rockvile1_final %>% 
+  get_unit_variables_r(construction_efficiency = .7,
+                     average_unit_size = 940) %>% 
+  group_by(housing_type) %>% 
+  count() %>% 
+  knitr::kable()
+
```

Comment:
units per lot coming out to 0 for SF?

Reply:
less than one will be rounded to 0 in the summary table. I added a new variable that's the unrounded unit per lot.